doc/learn/syntax-trees-with-typescript.md

```diff
@@ -0,0 +1,523 @@
+    // if we get here, node is Emphasis
+  }
+}
+
+if (is<Heading>(node, {type: 'heading', depth: 1})) {
+  // if we get here, node is Heading
+}
+
+// for advanced use cases, another predicate can be passed to `is`
+if (is<Literal>(node, (node: Node): node is Literal => "value" in node)) {
+  // if we get here, node is one of the Literal types
+}
+
+// reusable predicates can also be created using any `Test`
+const isBlockquote = convert('blockquote')
+if (isBlockquote(node)) {
+  // if we get here, node is Blockquote
+}
+```
+
+### How to build syntax tree
+
+When content needs to be created or added, it's often useful to build new syntax trees, or fragments of syntax trees.
+This can be easy to do with plain JSON, unified also offers some utilities for building trees with hyperscript or JSX.
+
+### JSON
+
+In many cases a tree can be created with plain JSON, for example:
+
+```ts
+const mdast = {
+  type: 'root',
+  children: [
+    {
+      type: 'paragraph',
+      children: [
+        {
+          type: 'text',
+          value: 'example'
+        }
+      ]
+    }
+  ]
+}
+```
+
+for some extra type safety this can be checked with the types for the given syntax tree language, in this case `mdast`:
+
+```ts
+import type {Root} from 'mdast'
+
+const mdast: Root = {
+  type: 'root',
+  children: [
+    {
+      type: 'paragraph',
+      children: [
+        {
+          type: 'text',
+          value: 'example'
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### `unist-util-builder`
+
+For more concise, hyperscript (or [`React.createElement`](https://reactjs.org/docs/react-api.html#createelement)) like syntax, [`unist-builder`](https://github.com/syntax-tree/unist-builder#readme) can be use:
+
+```ts
+import {u} from 'unist-builder'
+
+const mdast = u('root', [
+  u('paragraph', [
+    u('text', 'example')
+  ])
+])
+```
+
+#### `hastscript`
+
+For working with an HTML AST (HAST) it can be more familiar to work with [JSX](https://reactjs.org/docs/introducing-jsx.html), [`hastscript`](https://github.com/syntax-tree/hastscript#readme) provide familiar syntax:
+
+```tsx
+/* @jsxRuntime automatic */
+/* @jsxImportSource hastscript */
+/* @jsxFrag null */
+
+console.log(
+  <div class="foo" id="some-id">
+    <span>some text</span>
+    <input type="text" value="foo" />
+    <a class="alpha bravo charlie" download>
+      deltaecho
+    </a>
+  </div>
+)
+
+console.log(
+  <form method="POST">
+    <input type="text" name="foo" />
+    <input type="text" name="bar" />
+    <input type="submit" name="send" />
+  </form>
+)
+```
+
+#### `xastscript`
+
+Similarly for an XML AST (XAST). [`xastscript`](https://github.com/syntax-tree/xastscript#readme) provide familiar syntax:
+
+```tsx
+/* @jsxRuntime automatic */
+/* @jsxImportSource xastscript */
+/* @jsxFrag null */
+
+console.log(
+  <album id={123}>
+    <name>Born in the U.S.A.</name>
+    <artist>Bruce Springsteen</artist>
+    <releasedate>1984-04-06</releasedate>
+  </album>
+)
+```
```

Comment:
Guide abruptly ends. I’m missing somethings like Conclusion. Also a Further exercises / Further reading section would be good to have

Reply:
summary has been added